Adds the todos view page

Jonah-Northwood opened the pull request.

Jonah-Northwood committed:
Start todo list component

Just the beginnings of todo list component very unfinished.

Jonah-Northwood committed:
Create todo list component

Creates the files which allows for the viewing of the todo page as well as adding the paths to find them.

Jonah-Northwood committed:
Create todo service

Creates todo service and provides basic tests.

Jonah-Northwood committed:
Add filtering for body, category and owner

Body and category filter on client, owner filters on server. Structure for status filtering is in place but not working.
Cypress tests still need to be written for all.

LloydHilsgen001 reviewed (dismissed):
Well designed view page

Jonah-Northwood committed:
Fixed filtering by status

filteredTodos = filteredTodos.filter(todo => todo.status === filter.status); was not working and I am unsure why.
However, i converted both variables to strings and compared them and it works correctly now.

LloydHilsgen001 reviewed (approved):
Looks good!
